### PR TITLE
feat: add cdk8s completion spec

### PR DESCRIPTION
**Pull request by petar-cvit**
Adds [cdk8s](https://cdk8s.io/) completion spec

### CODE REVIEW

**Comment by petar-cvit** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/cdk8s.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        },
+      ],
+    },
+    {
+      name: "synth",
+      description:
+        "Synthesizes Kubernetes manifests for all charts in your app",
+      options: [
+        {
+          name: ["--app", "--a"],
```
Comment (fedeci) [inline src/cdk8s.ts:50]:
```suggestion
          name: ["--app", "-a"],
```

File: src/cdk8s.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+          description: "Command to use in order to execute cdk8s app",
+        },
+        {
+          name: ["--output", "-o"],
+          description: "Output directory",
```
Comment (fedeci) [inline src/cdk8s.ts:55]:
```suggestion
          description: "Output directory",
          args: {
            template: "folders",
            suggestCurrentToken: true,
          },
```

File: src/cdk8s.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          description:
+            "Write synthesized manifests to STDOUT instead of the output directory",
+        },
+        {
+          name: "--plugins-dir",
+          description: "Directory to store cdk8s plugins",
+        },
+        {
+          name: "--validate",
+          description: "Apply validation plugins on the resulting manifests",
```
Comment (fedeci) [inline src/cdk8s.ts:68]:
```suggestion
          description: "Apply validation plugins on the resulting manifests",
          exclusiveOn: ["--no-validate"],
```

File: src/cdk8s.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        },
+        {
+          name: "--no-validate",
+          description: "Disable validation",
```
Comment (fedeci) [inline src/cdk8s.ts:72]:
```suggestion
          description: "Disable validation",
          exclusiveOn: ["--validate"],
```

File: src/cdk8s.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+          description: "Does not add a prefix to generated class names",
+        },
+        {
+          name: "--exclude",
+          description:
+            "Do not import types that match these regular expressions",
+        },
+        {
+          name: ["--output", "-o"],
+          description: "Output directory",
```
Comment (fedeci) [inline src/cdk8s.ts:40]:
```suggestion
          description: "Output directory",
          args: {
            template: "folders",
            suggestCurrentToken: true,
          },
```

File: src/cdk8s.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        description: "Path to the CRD schema",
+      },
+      options: [
+        {
+          name: ["--language", "-l"],
+          description: "Output programming language",
+        },
+        {
+          name: "--class-prefix",
+          description: "A prefix to add to all generated class names",
```
Comment (fedeci) [inline src/cdk8s.ts:27]:
```suggestion
          description: "A prefix to add to all generated class names",
          exclusiveOn: ["--no-class-prefix"],
```

File: src/cdk8s.ts [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        },
+        {
+          name: "--no-class-prefix",
+          description: "Does not add a prefix to generated class names",
```
Comment (fedeci) [inline src/cdk8s.ts:31]:
```suggestion
          description: "Does not add a prefix to generated class names",
          exclusiveOn: ["--class-prefix"],
```

File: src/cdk8s.ts [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        },
+        {
+          name: "--exclude",
+          description:
+            "Do not import types that match these regular expressions",
```
Comment (fedeci) [inline src/cdk8s.ts:36]:
```suggestion
            "Do not import types that match these regular expressions",
          args: {
            name: "regexp",
          },
```

File: src/cdk8s.ts [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+        name: "type",
+        description: "Select language you are using",
+      },
+    },
+    {
+      name: "import",
+      description: "Import a CRD schema to generate generate resources",
+      args: {
+        name: "spec",
+        description: "Path to the CRD schema",
```
Comment (fedeci) [inline src/cdk8s.ts:18]:
```suggestion
        description: "Path to the CRD schema",
        args: {
           template: "filepaths",
        },
```

File: src/cdk8s.ts [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+          description: "Apply validation plugins on the resulting manifests",
+        },
+        {
+          name: "--no-validate",
+          description: "Disable validation",
+        },
+        {
+          name: "--validation-reports-output-file",
+          description:
+            "File to write a JSON representation of the validation reports to",
```
Comment (fedeci) [inline src/cdk8s.ts:77]:
```suggestion
            "File to write a JSON representation of the validation reports to",
          args: {
            template: "filepaths",
            suggestCurrentToken: true,
          },
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks for the spec @petar-cvit! I have added some arguments with generators as suggestions, they really make autocomplete powerful! I would suggest you to add some for the `--language` options too!

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**